app/src/Controller/SearchController.php

```diff
@@ -169,4 +124,26 @@ public function showAction(int $partitionTag, string $mailId, int $rid): Respons
             'message' => $ms,
         ]);
     }
+
+    private function sanitizeSortParameter(Request $request): void
+    {
+        $allowedSortFields = [
+            'mr.mailId',
+            'm.fromAddr',
+            'maddr.email',
+            'm.subject',
+            'm.timeIso',
+            'mr.bspamLevel',
+            'm.size',
+            'm.host',
+        ];
+
+        $sort = $request->query->getString('sort', 'm.timeIso');
+
+        if (!in_array($sort, $allowedSortFields, true)) {
+            $queryParams = $request->query->all();
+            $queryParams['sort'] = 'm.timeIso';
+            $request->query->replace($queryParams);
```

Comment:
On peut pas juste faire :


```suggestion
            $request->query->set('sort', 'm.timeIso');
```

?

Reply:
Effectiveme
nt, c'est plus simple